src/bin/pgcopydb/table-data.c

```diff
@@ -1182,12 +1183,21 @@ copydb_prepare_copy_query(CopyTableDataSpec *tableSpecs,
 
 		for (int i = 0; i < table->attributes.count; i++)
 		{
+			/* Generated columns cannot be used in COPY */
+			if (table->attributes.array[i].attisgenerated)
+			{
+				continue;
+			}
+
 			char *attname = table->attributes.array[i].attname;
```

Comment:
Nitpicking: now that we access to more than one slot of the structure instance, could you maybe declare a new local variable and use it, as in:

``` c
    SourceTableAttribute *attribute = &(table->attributes.array[i]);
```

And then use `attribute->attname` and attribute->attisgenerated`.